shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/documentservice/ExecutionDocumentService.java

```diff
@@ -127,17 +123,139 @@ private Document getUpdateOperationOnFullExecutionData(String executionData) thr
+        return dbObjectInput;
+    }
+
+    private Document getSearchFilterForGraph_EndPointDetails(Integer clientId){
+        Document dbObjectInput = new Document();
+        dbObjectInput.append(ExecutionDocumentConstants.Fields.CLIENT_ID, clientId);
+        return dbObjectInput;
+    }
+
     private MongoClient getMongoClient() {
         String mongoConnectionUri = ExecutionDocumentServiceHelper.createMongoConnectionUri(this.shepherdConfiguration.getDataSourceDetails());
         MongoClientURI uri = new MongoClientURI(mongoConnectionUri);
         return new MongoClient(uri);
     }
+
+    public boolean updateEndPointDetails(EndpointDetails endpointDetails) throws IOException {
+        if(this.mongoClient == null){
+            this.mongoClient = getMongoClient();
+        }
+        try {
+            MongoCollection<Document> collection = ExecutionDocumentServiceHelper.getMongoCollection(this.mongoClient, "shepherd_graph_endpoint", "graph_endpoint");
+
+            if (collection != null) {
+                UpdateResult updateResult = collection.updateOne(getSearchFilterForGraph_EndPointDetails(endpointDetails.getClientId(),endpointDetails.getEndpointName()), getUpdateOperationOnFullGraphDetails(endpointDetails), new UpdateOptions());
+                log.debug("updateDocument() :: database: " + "shepherd_graph_endpoint" + " and collection: " + "graph_endpoint"
+                        + " is document Updated :" + updateResult.wasAcknowledged());
+                boolean ack = updateResult.wasAcknowledged();
+                return ack;
+            }
+        } catch (MongoWriteException e) {
+            log.error(e.getMessage(), e);
+            throw e;
+        } catch (Exception ex) {
+            throw ex;
+        }
+        return false;
+    }
+
+    public ObjectId registerEndPointDetails(EndpointDetails endpointDetails) throws IOException {
+        if(this.mongoClient == null){
+            this.mongoClient = getMongoClient();
+        }
+        try {
+            MongoCollection<Document> collection = ExecutionDocumentServiceHelper.getMongoCollection(this.mongoClient, "shepherd_graph_endpoint", "graph_endpoint");
+
+            if (collection != null) {
+                final Document dbObjectInput = new Document();
+
+                dbObjectInput.append(ExecutionDocumentConstants.Fields.CLIENT_ID,endpointDetails.getClientId());
+                dbObjectInput.append(ExecutionDocumentConstants.Fields.ENDPOINT_NAME, endpointDetails.getEndpointName());
+                dbObjectInput.append(ExecutionDocumentConstants.Fields.ENDPOINT_DETAILS, JSONLoader.stringify(endpointDetails));
+                collection.insertOne(dbObjectInput);
+                log.debug(String.format("EndPoint details are inserted : %s inserted successfully in \n collection : %s and db : %s", endpointDetails, "graph_endpoint", "shepherd_graph_endpoint"));
```

Comment:
Remove all the hardcoded strings. Please make this habit from day#1.  Use root level constants file for the same.